### DISCUSSION TITLE
Potoprism unexpectedly collapsed

### DISCUSSION THREAD

**Discussion by ai30328348** [photoprism/photoprism#4261] [Q&A] [answered]
Crashes when scanning pictures larger than 100MB due to high CPU usage

photoprism-1  | time="2024-05-16T03:16:51Z" level=info msg="media: 'mz20pro.png' was taken at 2024-04-25 15:01:15.23345566 +0000 UTC (file mod time)"
photoprism-1  | /scripts/entrypoint.sh: line 103:  1323 Killed                  /usr/bin/setpriv --reuid "${PHOTOPRISM_UID}" --regid "${PHOTOPRISM_GID}" --init-groups --inh-caps -all "$@"
mariadb-1     | 2024-05-16  3:21:50 6 [Warning] Aborted connection 6 to db: 'photoprism' user: 'photoprism_sql' host: '172.19.0.3' (Got an error reading communication packets)
mariadb-1     | 2024-05-16  3:21:50 9 [Warning] Aborted connection 9 to db: 'photoprism' user: 'photoprism_sql' host: '172.19.0.3' (Got an error reading communication packets)
mariadb-1     | 2024-05-16  3:21:50 4 [Warning] Aborted connection 4 to db: 'photoprism' user: 'photoprism_sql' host: '172.19.0.3' (Got an error reading communication packets)
mariadb-1     | 2024-05-16  3:21:50 3 [Warning] Aborted connection 3 to db: 'photoprism' user: 'photoprism_sql' host: '172.19.0.3' (Got an error reading communication packets)
mariadb-1     | 2024-05-16  3:21:50 5 [Warning] Aborted connection 5 to db: 'photoprism' user: 'photoprism_sql' host: '172.19.0.3' (Got an error reading communication packets)
mariadb-1     | 2024-05-16  3:21:50 8 [Warning] Aborted connection 8 to db: 'photoprism' user: 'photoprism_sql' host: '172.19.0.3' (Got an error reading communication packets)
mariadb-1     | 2024-05-16  3:21:50 7 [Warning] Aborted connection 7 to db: 'photoprism' user: 'photoprism_sql' host: '172.19.0.3' (Got an error reading communication packets)
photoprism-1 exited with code 0

**Comment by graciousgrey** [comment, marked as answer]
Does your server fulfill our system requirements? Do you have at least 4GB Swap configured?

https://docs.photoprism.app/getting-started/#system-requirements

**Comment by ai30328348** [comment]
emmm...
At that time, I'm verifying the rar file, which may have taken up too much memory and CPU.
Only the photoprism stopped working, the mariadb is still working

**Reply by ai30328348** [reply]
emmmm。。。。
Photoprism doesn't limit the CPU and memory, and it crashes when the image is large enough

**Reply by lastzero** [reply]
Yes, that's right. See our documentation and Getting Started FAQ for details, e.g.:
- https://docs.photoprism.app/getting-started/faq/#why-is-my-configured-memory-limit-exceeded-when-indexing-even-though-photoprism-doesnt-actually-seem-to-use-that-much-memory

We could implement paging in our application, but your operating system does it faster and better. However, I have just added support for thumbnail generation with libvips, which you are welcome to test and give us feedback on:
- https://github.com/photoprism/photoprism/issues/1474

It is supposed to require less memory and thus might solve your issue!

**Reply by lastzero** [reply]
@ai30328348 any update on this?

**Reply by ai30328348** [reply]
No, after I redeployed, the scan didn't crash.

**Reply by lastzero** [reply]
So we can consider the issue to be fixed?

**Reply by ai30328348** [reply]
I didn't see any cases of high memory usage in the preview version, and it looks like the memory usage is limited.
How do I close this discussion?
